catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/transformer/TransformerDescriptorUtilities.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.transformer;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.osgi.framework.ServiceReference;
+
+public class TransformerDescriptorUtilities {
```

Comment:
I think a lot of your code could be simpler if this wasn't a static utility class. If this class was non-static then you could instantiate it in your blueprint and inject the reference lists of query response transformers and metacard transformers and only this object would need to know about them. This class would have to have separate methods for query response transformers and metacard transformers, but then your other classes don't have to bother with passing around lists of service references.

Maintaining references to blueprint's reference lists here and iterating over them instead will also allow this code to properly handle transformers being removed or added at runtime. Currently your lists of maps will be set once when their respective objects are instantiated and their setters are called with the _currently available_ transformers. If transformers are removed or added then those lists of maps will not be updated. Making this change will allow everything to be done dynamically through this class and you won't have to maintain separate lists of maps.